### PR TITLE
client api code

### PR DESCRIPTION
**Pull request by sanidhya290**

### CODE REVIEW

File: controllers/clientController.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
+const Client = require('../models/client');
+const upload = require('../middleware/multer'); // Import multer configuration
+const mongoose = require('mongoose');
+exports.registerClient = async (req, res) => {
+    upload(req, res, async (err) => {
```
Comment (anujtembhurne-ai) [inline controllers/clientController.js]: Add upload related code in routes

File: controllers/clientController.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        if (err) {
+            return res.status(400).json({ error: err.message });
+        }
+
+        try {
+            // Create a new client object with file paths
+            const client = new Client({
+                dealerName: req.body.dealerName,
```
Comment (anujtembhurne-ai) [inline controllers/clientController.js]: change this into proper format

File: controllers/clientController.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+            // Save the client to the database
+            await client.save();
+            res.status(201).json(client);
+        } catch (error) {
+            res.status(400).json({ error: error.message });
+        }
+    });
+};
+
+exports.getAllClients = async (req, res) => {
```
Comment (anujtembhurne-ai) [inline controllers/clientController.js]: CHange export line and add it into last line, export all apis there

File: controllers/clientController.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+                coverLetterFile: req.files['coverLetterFile'] ? req.files['coverLetterFile'][0].path : '',
+                productCertFile: req.files['productCertFile'] ? req.files['productCertFile'][0].path : '',
+                isoCertFile: req.files['isoCertFile'] ? req.files['isoCertFile'][0].path : '',
+                brochureFile: req.files['brochureFile'] ? req.files['brochureFile'][0].path : '',
+                technicalSpecification:req.body.technicalSpecification
+            });
+
+            // Save the client to the database
+            await client.save();
+            res.status(201).json(client);
```
Comment (anujtembhurne-ai) [inline controllers/clientController.js]: add return keyword in the starting of this line

File: controllers/clientController.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        } catch (error) {
+            res.status(400).json({ error: error.message });
```
Comment (anujtembhurne-ai) [inline controllers/clientController.js]: 500 status code needs to be added

**Review by anujtembhurne-ai — COMMENTED** [review]
Please make the mentioned changes

File: controllers/clientController.js (hunk 3 above)
Comment (sanidhya290) [reply controllers/clientController.js]: done

File: controllers/clientController.js (hunk 4 above)
Comment (sanidhya290) [reply controllers/clientController.js]: done

File: controllers/clientController.js (hunk 5 above)
Comment (sanidhya290) [reply controllers/clientController.js]: done

File: controllers/clientController.js (hunk 1 above)
Comment (sanidhya290) [reply controllers/clientController.js]: done

**Pull request merged**